packages/std/src/memory.rs

```diff
@@ -64,9 +74,18 @@ pub unsafe fn consume_region(ptr: *mut c_void) -> Result<Vec<u8>, Error> {
 /// the resulting data.
 /// The Box must be dropped (with scope), but not the data
 pub fn build_region(data: &[u8]) -> Box<Region> {
+    let data_ptr = data.as_ptr() as usize;
+    build_region_from_components(
+        u32::try_from(data_ptr).expect("pointer doesn't fit in u32"),
```

Comment:
definitely safer than `as` :+1:

Reply:
I ran into crashes when trying to test this code on my 64 bit machine, where pointers don't fit into u32. Then I created this explicit conversion code and understood that it is not possible to unit test memory.rs on non-32bit platforms.